src/sysdir.c

```diff
@@ -85,8 +85,12 @@ static int git_sysdir_guess_template_dirs(git_buf *out)
 
 typedef int (*git_sysdir_guess_cb)(git_buf *out);
 
-static git_buf git_sysdir__dirs[GIT_SYSDIR__MAX] =
-	{ GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT };
+struct git_sysdir__dir {
+	int initialized;
```

Comment:
Nope, I'm wrong.  I didn't see the `git_sysdirs_global_init` being called by `git_libgit2_init`.

You're right:  this has become fairly overwrought.